code/modules/mob/living/silicon/silicon.dm

```diff
@@ -59,11 +59,12 @@
 	. = ..()
 	GLOB.silicon_mobs += src
 	faction += "silicon"
-	for(var/datum/atom_hud/data/diagnostic/diag_hud in GLOB.huds)
-		diag_hud.add_to_hud(src)
-	diag_hud_set_status()
-	diag_hud_set_health()
-	add_sensors()
+	if(sensors_on)
```

Comment:
this will break diag huds if silicons don't have sensors_on at init.

Make a new var for diag hud

Reply:
wdym